ee/query-service/license/db.go

```diff
@@ -49,15 +50,18 @@ func (r *Repo) GetLicenses(ctx context.Context) ([]model.License, error) {
 }
 
 // GetActiveLicense fetches the latest active license from DB
-func (r *Repo) GetActiveLicense(ctx context.Context) (*model.License, error) {
+func (r *Repo) GetActiveLicense(ctx context.Context) (*model.License, *basemodel.ApiError) {
 	var err error
 	licenses := []model.License{}
 
 	query := "SELECT key, activationId, planDetails, validationMessage FROM licenses"
 
 	err = r.db.Select(&licenses, query)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get active licenses from db: %v", err)
+		if err == sql.ErrNoRows {
+			return nil, basemodel.NotFoundError(errors.New("no licenses found"))
+		}
```

Comment:
The `Select` never returns the `sql.ErrNoRows` so this change doesn't add anything. Some call paths expect a nil error when there is no license instead of a NotFound error otherwise, the server won't start https://github.com/SigNoz/signoz/blob/93df4759699025aaf328f18e00e8e0d758a06553/ee/query-service/main.go#L154. Either we should handle the no license error wherever GetActiveLicense is involved or return nil error when there is no active licence.

Reply:
- Why won't Select return a NoRows error? There can be cases where no license is there right?
 - Not sure whether the above is happening or not. (testing had no license, server started). If an active license is not found, the default is set to BASIC as per my understanding. 
 
 Ref line 123 in `[ee/query-service/license/manager.go](https://github.com/SigNoz/signoz/pull/4762/files#diff-571d493907b4315fc1ac269efcf7668db417a5a591c6cd62e582046ed24b1043)`:
 ```
if active != nil {
		lm.SetActive(active)
	} else {
		zap.L().Info("No active license found, defaulting to basic plan")
		// if no active license is found, we default to basic(free) plan with all default features
		lm.activeFeatures = model.BasicPlan
		setDefaultFeatures(lm)
```

Reply:
>Why won't Select return a NoRows error? There can be cases where no license is there, right?

The [`Get`](https://pkg.go.dev/github.com/jmoiron/sqlx@v1.3.4#DB.Get) returns the `NoRows` error but [`Select`](https://pkg.go.dev/github.com/jmoiron/sqlx@v1.3.4#Select) doesn't.

> If an active license is not found, the default is set to BASIC as per my understanding.

If you see the same function, it returns the error on line 121 if it's non-nil. The default to the basic plan happens only when the err is nil and the active plan is nil. There is no issue right now because `Select` doesn't return NoError but I think we should handle it?